src/hotspot/cpu/x86/x86.ad

```diff
@@ -8681,13 +8672,13 @@ instruct vpopcount_integral_reg_evex_masked(vec dst, vec src, kReg mask) %{
 %}
 
 instruct vpopcountI_avx_reg(vec dst, vec src, vec xtmp1, vec xtmp2, rRegP rtmp) %{
-  predicate((!VM_Version::supports_avx512_vpopcntdq() && Matcher::vector_element_basic_type(n->in(1)) == T_INT) ||
+  predicate(!UsePopCountInstruction ||
+            (!VM_Version::supports_avx512_vpopcntdq() && Matcher::vector_element_basic_type(n->in(1)) == T_INT) ||
             (!VM_Version::supports_avx512_bitalg() && is_subword_type(Matcher::vector_element_basic_type(n->in(1)))));
```

Comment:
Could be replaced by:
predicate((Matcher::vector_element_basic_type(n->in(1)) != T_LONG) 
 && !is_pop_count_instr_target(Matcher::vector_element_basic_type(n->in(1))));